### ISSUE TITLE
🔥[Community Event] Doc Tests Sprint - Configuration files🔥

### ISSUE THREAD

**Issue by ydshieh** [huggingface/transformers#19487]
This sprint is similar to #16292 - but for model **configuration files**, i.e. `configuration_[model_name].py`.
For example, `src/transformers/models/bert/configuration_bert.py` 

# The expected changes

The changes we expect  could be find #19485:

  1. **Change the import order of the model and configuration classes**
  2. **Add `(with random weights)` in the  comment before model initialization line**
  3. **Add `configuration_[model_name].py` to `utils/documentation_tests.txt`** (respecting the order)

Please do step 3. only after **Running the doctest and make sure all tests pass** (see below) 🙏 

# How to run doctests

Suppose you are working on `src/transformers/models/bert/configuration_bert.py`. The steps to run the test are:

0. **Stage your changes**

    ```bash
    git add src/transformers/models/bert/configuration_bert.py
    ```
1. **Prepare the files to be tested**

    ```python
    python utils/prepare_for_doc_test.py src
    ```
    or if you prefer to be more specific 
    ```python
    python utils/prepare_for_doc_test.py src/transformers/models/bert/configuration_bert.py
    ```
    This will change some files (doc-testing needs to add additional lines that we don't include in the doc source files).
2. **Launch the test:**
    ```python
    python -m pytest --doctest-modules src/transformers/models/bert/configuration_bert.py -sv --doctest-continue-on-failure
    ```
3. **Cleanup git status**
    ```bash
    git checkout -- .
    ```
    to clean up the changes in step 1.

# Ready (or not)?

If all tests pass, you can commit, push and open a PR 🔥 🚀 , otherwise iterate the above steps 💯 !

**Comment by daspartho** [comment]
I'd like to work on this; I'll start with YOLOS and open a PR :)

**Comment by daspartho** [comment]
I'll take on Whisper!

**Comment by daspartho** [comment]
I'll work on Beit!

**Comment by imarekkus** [comment]
FYI Bart #19524 and Albert #19541 are already done :)

**Comment by imarekkus** [comment]
If you don't mind I will work on [big_bird](https://github.com/imarekkus/transformers/tree/main/src/transformers/models/big_bird) and [bigbird_pegasus](https://github.com/imarekkus/transformers/tree/main/src/transformers/models/bigbird_pegasus) models.

**Comment by daspartho** [comment]
I'll take on GPT2 next!

**Comment by grgkaran03** [comment]
I'll take yoso

**Comment by grgkaran03** [comment]
i'll work on yoso

**Comment by daspartho** [comment]
I'll work on
- RoBERTa
- ViT
- DeiT
- Reformer

and open up PR soon :)

**Comment by RamitPahwa** [comment]
Also will raise for Transformer-XL !

**Comment by imarekkus** [comment]
I'll work on bloom

**Comment by imarekkus** [comment]
I'll work on ctrl

**Comment by SD-13** [comment]
Hi @ydshieh, I am new here and want to contribute to this issue. Can you please help me to find the remaining files? Thanks,

**Comment by ydshieh** [comment]
> Hi @ydshieh, I am new here and want to contribute to this issue. Can you please help me to find the remaining files? Thanks,

Hi @SD-13 You can check the latest file 
https://github.com/IzicTemi/transformers/blob/main/utils/documentation_tests.txt

Any configuration file that is not on that list (on `main` branch) **and [not claimed here yet, no PR opened yet]** are all welcome :-)

**Comment by IzicTemi** [comment]
Hey @ydshieh, could I fix multiple models in a single PR, or do I have to open a single PR for each fix?

**Comment by grgkaran03** [comment]
I will be taking blenderbots as well

**Comment by SD-13** [comment]
Hey @ydshieh, I am getting this error
![Screenshot from 2022-10-13 19-44-43](https://user-images.githubusercontent.com/89520981/195621668-e2ba4aee-dee6-45a3-9699-b7b61482bfe6.png)
can you please help me to understand the reason and fix it? Thanks,

**Comment by ydshieh** [comment]
@SD-13 Could you open a PR (even if it is not complete yet), and post the command you run and the complete error message in that PR? Using image is not very convivence to search and debug 🙏

**Comment by SD-13** [comment]
> @SD-13 Could you open a PR (even if it is not complete yet), and post the command you run and the complete error message in that PR? Using image is not very convivence to search and debug pray

Yeah true. I created [this](https://github.com/huggingface/transformers/pull/19580) PR and let's leave those errors since all checks are passing. Thanks,

**Comment by SD-13** [comment]
I will also take `time_series_transformer.py`, `vision_encoder_decoder.py`, and `trajectory_transformer.py`. Thanks,

**Comment by grgkaran03** [comment]
took up blenderbot_small too.

**Comment by daspartho** [comment]
I'll take on
- RoBERTa
- ViT
- DeiT
- Reformer

and will open up PR :)

**Comment by daspartho** [comment]
I'll take on
- SEW
- SEW-D
- Swin
- Swin V2
- UniSpeech

and will open up PR :)

**Comment by SaurabhBudhwani26** [comment]
Hi can you assign me CANINE and CTRL?

**Comment by SaurabhBudhwani26** [comment]
Hi @ydshieh , I want to contribute to this issue. Can you please help me to find the remaining files?

**Comment by SD-13** [comment]
> Hi @ydshieh , I want to contribute to this issue. Can you please help me to find the remaining files?

Hey @SaurabhBudhwani26 , please check [this](https://github.com/huggingface/transformers/issues/19487#issuecomment-1277518647). I think it will be helpful. Thanks,

**Comment by ztjhz** [comment]
I'll work on `configuration_visual_bert.py`

**Comment by Xabilahu** [comment]
I will work on:
- `big_bird`
- `bigbird_pegasus`

**Comment by ztjhz** [comment]
I will work on
- `configuration_xlm_roberta.py`
- `configuration_xlm_roberta_xl.py`

**Comment by AShreyam** [comment]
Can I work on this?
@ydshieh

**Comment by AShreyam** [comment]
I will work on **`flava`**

**Comment by AShreyam** [comment]
Now, I will work on `Longformer`, `Pegasus` and `VisualBERT`.

**Comment by ydshieh** [comment]
@AShreyam You didn't open a PR, and you merged your branch into your own main branch.

**Comment by AShreyam** [comment]
I am really sorry for the inconvenience.

**Comment by AShreyam** [comment]
Here is the actual pull request. Thank You :)

**Comment by lappemic** [comment]
Work on `configuration_big_bird.py`

**Comment by daspartho** [comment]
I'll take on
- LeViT
- DistilBERT
- ResNet

and open up PR :)

**Comment by AymenBer99** [comment]
I'll take on CodeGen

**Comment by daspartho** [comment]
I'll take on Data2Vec ones :)

**Comment by AymenBer99** [comment]
I'll take on conditional_detr

**Comment by ak04p** [comment]
I'll take on realm

**Comment by AymenBer99** [comment]
I'll take on convbert

**Comment by daspartho** [comment]
I'll take on CLIP =)

**Comment by AymenBer99** [comment]
I'll take XLNet

**Comment by juancopi81** [comment]
I can take the wav2vec2 ones if that is ok, is someone else working on those?

**Comment by AymenBer99** [comment]
I'll take XLM

**Comment by AymenBer99** [comment]
I'll take cvt

**Comment by mukesh663** [comment]
I'll take `pegasus` and `pegasus_x`

**Comment by mukesh663** [comment]
I'll take `pegasus` and `pegasus_x`

**Comment by soma2000-lang** [comment]
@ydshieh Added config files for convnext

**Comment by ydshieh** [comment]
Oh it will close this ...?

**Comment by ndrohith09** [comment]
I would take `gpt_neo` and `gpt_neox`

**Comment by daspartho** [comment]
I'll take on
- SpeechToTextTransformer
- SpeechToTextTransformer2
- SqueezeBERT

**Comment by ndrohith09** [comment]
I would like to work on `openai` and `opt`

**Comment by Revanth2002** [comment]
I will take `mbart` and `mctct`

**Comment by Revanth2002** [comment]
I will work on `layoutlm` , `layoutlmv2` , `layoutlmv3`

**Comment by ayaka14732** [comment]
I will work on ELECTRA

**Comment by ayaka14732** [comment]
I will work on PoolFormer

**Comment by ayaka14732** [comment]
I will work on PLBART

**Comment by ayaka14732** [comment]
I will work on Nezha

**Comment by sha016** [comment]
I'll take maskformer

**Comment by rushic24** [comment]
Hi, Can I have LayoutLMv2 and BERT

**Comment by ydshieh** [comment]
> Hi, Can I have LayoutLMv2 and BERT

Hi @rushic24  They have been done. You can check [this file](https://github.com/huggingface/transformers/blob/main/utils/documentation_tests.txt) and find other config files to work with 🤗

**Comment by sha016** [comment]
I'll take fsmt next

**Comment by Saad135** [comment]
While browsing the list of model configurations, I noticed that the DebertaConfig class does not have an example docstring section. Unsure if that is supposed to be like that, but just incase its not, I will add a PR to include the example docstring and maybe I can get some feedback from there.

**Comment by ydshieh** [comment]
> DebertaConfig

That would be very nice, @Saad135 ! Thank you

**Comment by Saad135** [comment]
I will take DeBERTa-v2 next

**Comment by Saad135** [comment]
I can take camembert next

**Comment by Saad135** [comment]
I can take DPR next

**Comment by Saad135** [comment]
I can take DeformableDetrConfig next

**Comment by JuheonChu** [comment]
Can I take timesformer next?

**Comment by ydshieh** [comment]
@elabongaatuo GPT-J is large, and our CI won't be able to run doctest with its checkpoints. 

I think gptj, longformer, and hubert are all covered in 

https://github.com/gante/transformers/blob/eb4493f352ea257373a957ba503e3b3d842a37ca/utils/documentation_tests.txt
https://github.com/huggingface/transformers/blob/eb4493f352ea257373a957ba503e3b3d842a37ca/utils/documentation_tests.txt
https://github.com/amyeroberts/transformers/blob/eb4493f352ea257373a957ba503e3b3d842a37ca/utils/documentation_tests.txt

Feel free to check the modeling files that are not in the above file 🤗 if you want to work on it ❤️ . Thank you!

**Comment by elabongaatuo** [comment]
@ydshieh , thank you. m2m_100,llama and mvp don't have modeling files. a go ahead to work on them?

**Comment by ydshieh** [comment]
`llama` has no publicly available checkpoints on the Hub - no need to work on it.
For the other 2 files, you can run doctest against them. If they pass, you can simply add them to `documentation_tests.txt`.
Othewise, we can discussion how to deal with the errors :-)

**Comment by PhalitJotwani** [comment]
Hi @ydshieh , I am new to open source, so just wanted to confirm whether I can take Falcon or not? Config file of Falcon is not mentioned in the documentation_tests.txt file.

**Comment by AVAniketh0905** [comment]
Hello @ydshieh , I am new to open sorce and want to take `barthez`. If my contributions are successful, I'm eager to extend my involvement to other models as well. Looking forward to a productive and enduring journey of contributions!

**Comment by Adithya4720** [comment]
I'll take roformer
